testers/SporadicNaN.cpp

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2020 Nicholas Corgan
+// SPDX-License-Identifier: BSL-1.0
+
+#include <Pothos/Framework.hpp>
+
+#include <limits>
+#include <random>
+
+template <typename T>
+class SporadicNaN: public Pothos::Block
+{
+public:
+    SporadicNaN():
+        _gen(_rd()),
+        _randomProb(0.0, 1.0),
+        _probability(0.0)
+    {
+        using Class = SporadicNaN<T>;
+
+        static const Pothos::DType dtype(typeid(T));
+        
+        this->setupInput(0, dtype);
+        this->setupOutput(0, dtype, this->uid()); // Unique domain because of buffer forwarding
+        this->registerCall(this, POTHOS_FCN_TUPLE(Class, getProbability));
+        this->registerCall(this, POTHOS_FCN_TUPLE(Class, setProbability));
+    }
+
+    double getProbability(void) const
+    {
+        return _probability;
+    }
+
+    void setProbability(double prob)
+    {
+        if((prob < 0.0) || (prob > 1.0))
+        {
+            throw Pothos::RangeException(
+                "SporadicNaN::setProbability("+std::to_string(prob)+")",
+                "probability not in [0.0, 1.0]");
+        }
+        _probability = prob;
+    }
+
+    void work() override
+    {
+        if(0 == this->workInfo().minInElements)
+        {
+            return;
+        }
+
+        auto inputPort = this->input(0);
+        auto outputPort = this->output(0);
+
+        // Calculate if a NaN will be injected.
+        const bool insertNaN = (_randomProb(_gen) <= _probability);
+
+        const auto &buffer = inputPort->takeBuffer();
```

Comment:
this block is changing an input buffer, not just forwarding it or a part of it. cant do that, its not safe

Reply:
Done.